### PR TITLE
Dynamic queries

### PR DESCRIPTION
**Pull request by kevin-bruhwiler**
@menuka94 Can you take a look at this and tell me if you see any obvious issues?

### CODE REVIEW

**Comment by kevin-bruhwiler** [comment]
@menuka94 still waiting for you to take a look at this. Also, is there a limit on the number of open grpc requests? After the client makes 6 or so, using my new code, the server will stop accepting them for a while. I'm worried they're not being closed properly.

File: src/main/proto/sustain.proto [hunk 1]
```diff
@@ -158,3 +160,48 @@ message Predicate {
...
+        CompoundRequest second_compound_request = 5;
+    }
+}
+
+
+message Query {
+    string host = 1;
+    int32 port = 2;
+    string collection = 3;
+    string query = 4; //This is a json representation of a mongodb query
```
Comment (menuka94) [inline src/main/proto/sustain.proto:206]: Do you plan on including examples for this?

**Comment by menuka94** [comment]
> @menuka94 still waiting for you to take a look at this. Also, is there a limit on the number of open grpc requests? After the client makes 6 or so, using my new code, the server will stop accepting them for a while. I'm worried they're not being closed properly.

Sorry, I had a couple of assignments to submit today.

**Comment by kevin-bruhwiler** [comment]
There definitely is. I managed so solve some issues by manually closing the streams on the client-side, but that doesn't help in instances where I need to run many queries simultaneously. I can verify that the stream "end" event is never being called, which means streams are staying open forever. This is an issue, because the client doesn't know when a stream is finished, so doesn't know when to close it. How do you close a stream from the server side?

File: src/main/proto/sustain.proto (hunk 1 above)
Comment (kevin-bruhwiler) [reply src/main/proto/sustain.proto:206]: What do you mean? There are a couple examples on the client-side already.

**Comment by kevin-bruhwiler** [comment]
Ok, I figured out what the issue was with my streams not getting closed. If you're alright with these changes, can you approve them?

**Review by menuka94 — APPROVED** [review]

File: src/main/proto/sustain.proto [hunk 2]
```diff
@@ -4,7 +4,7 @@ import "google/api/http.proto";
 
 option java_multiple_files = true;
 option java_package = "org.sustain";
-option java_outer_classname = "SustainQueryService";
+option java_outer_classname = "SustainService";
```
Comment (menuka94) [inline src/main/proto/sustain.proto:7]: Can you explain why this was changed?

Comment (kevin-bruhwiler) [reply src/main/proto/sustain.proto]: I think that was something I tried when I couldn't get the gRPC queries to be recognized. I've changed it back

**Pull request merged**